rust/src/interpreter/utils.rs

```diff
@@ -0,0 +1,127 @@
+use std::cmp::min;
+
+use evmc_vm::{ExecutionMessage, MessageFlags, Revision, StatusCode};
+
+use crate::{interpreter::gas::consume_copy_cost, types::u256};
+
+#[inline(always)]
+pub fn get_mut_slice_within_bounds<T>(data: &[T], offset: u256, len: u64) -> &[T] {
+    if len == 0 {
+        return &[];
+    }
+    let (offset, offset_overflow) = offset.into_u64_with_overflow();
+    if offset_overflow {
+        return &[];
+    }
+    let offset = offset as usize;
+    let len = len as usize;
+    let (end, end_overflow) = offset.overflowing_add(len);
+    if end_overflow || offset >= data.len() {
+        &[]
+    } else {
+        &data[offset..min(end, data.len())]
+    }
+}
+
+#[inline(always)]
+pub fn zero_slice(data: &mut [u8]) {
```

Comment:
```suggestion
pub fn set_to_zero(data: &mut [u8]) {
```
Functions should have a verb, a noun would make one think that this function returns the zero value. 
extra mile: could be a trait. 
(exceptions in the language are things like from, into, new.... but those have well defined semantics)